src/buildHTML.js

```diff
@@ -71,23 +103,25 @@ export const buildExpression = function(expression, options, isRealGroup,
 
     // Ignore explicit spaces (e.g., \;, \,) when determining what implicit
     // spacing should go between atoms of different classes, and add dummy
-    // spans for determining spacings between surrounding atoms
-    const nonSpaces = [
-        surrounding[0] && makeSpan([surrounding[0]], [], options),
+    // spans for determining spacings between surrounding atoms.
+    const nonSpaces: (?HtmlDomNode)[] = [
+        surrounding[0] ? makeSpan([surrounding[0]], [], options) : null,
         ...rawGroups.filter(group => group && group.classes[0] !== "mspace"),
-        surrounding[1] && makeSpan([surrounding[1]], [], options),
+        surrounding[1] ? makeSpan([surrounding[1]], [], options) : null,
     ];
 
     // Before determining what spaces to insert, perform bin cancellation.
     // Binary operators change to ordinary symbols in some contexts.
     for (let i = 1; i < nonSpaces.length - 1; i++) {
-        const left = getOutermostNode(nonSpaces[i], "left");
+        // $FlowFixMe: Only the first and last can be nullable.
+        const nonSpacesI: HtmlDomNode = nonSpaces[i];
```

Comment:
`$FlowFixMe` just allows the assignment; flow would still type `nonSpaces` as `HtmlDomNode` and assume its type as that in the subsequent lines. In any case, I added a general-purpose `assert()` method in utils since I've wanted it more than once before and removed the `$FlowFixMe`.